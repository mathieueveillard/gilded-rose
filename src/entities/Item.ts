class Item {
  private _name: string;
  private _sellIn: number;
  private _quality: number;

  constructor(name: string, sellIn: number, quality: number) {
    this._name = name;
    this._sellIn = sellIn;
    this._quality = quality;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get sellIn(): number {
    return this._sellIn;
  }

  set sellIn(value: number) {
    this._sellIn = value;
  }

  get quality(): number {
    return this._quality;
  }

  set quality(value: number) {
    this._quality = value;
  }
}

export default Item;
