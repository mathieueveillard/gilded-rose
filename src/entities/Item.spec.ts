import runGoldenMaster from 'jest-golden-master';

import Item from './Item';

describe('Item', () => {
  it('should initialize correctly', () => {
    runGoldenMaster(async () => {
      const item = new Item('Test Item', 10, 20);
      expect(item.name).toBe('Test Item');
      expect(item.sellIn).toBe(10);
      expect(item.quality).toBe(20);
    });
  });

  it('should set and get properties correctly', () => {
    runGoldenMaster(async () => {
      const item = new Item('Test Item', 10, 20);
      item.name = 'New Test Item';
      item.sellIn = 5;
      item.quality = 25;
      expect(item.name).toBe('New Test Item');
      expect(item.sellIn).toBe(5);
      expect(item.quality).toBe(25);
    });
  });
});
