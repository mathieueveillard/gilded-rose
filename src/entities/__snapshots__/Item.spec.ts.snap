// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should initialize correctly 1`] = `[]`;

exports[`Item should set and get properties correctly 1`] = `[]`;
